ir/instr.cpp

```diff
@@ -231,6 +232,14 @@ static StateValue fm_poison(State &s, expr a, expr b,
   return { move(val), move(non_poison) };
 }
 
+static StateValue fm_poison(State &s, expr a, expr b,
+                            function<expr(expr&,expr&)> fn,
+                            FastMathFlags fmath, bool only_input) {
+  return fm_poison(s, a, b, expr(), 
```

Comment:
expr() doesn't work here. It will make all non_poison expressions above to be null.

Reply:
Fixed, but kept this the same. I experimented passing expr(true) but that gets simplified and gives the same result as expr(). Instead the passed expr c is tested with isValid() before doing any operation with it, and if expr() is passed, nothing is added to the non_poison expr.

Reply:
That doesn't work because c might be null.
You need an extra argument, say "bool is_ternary = true".

Reply:
Also, in this line you need move() for a & b.

Reply:
Thanks!, all fixed, i need to pay more attention.